src/main/scala/org/neo4j/spark/v2/Neo4jOptions.scala

```diff
@@ -0,0 +1,107 @@
+package org.neo4j.spark.v2
+
+import org.neo4j.driver.Config.TrustStrategy
+
+class Neo4jOptions(parameters: java.util.Map[String, String]) {
+
+  import Neo4jOptions._
+  import QueryType._
+
+  private def getRequiredParameter(parameter: String): String = {
+    if (!parameters.containsKey(parameter) || parameters.get((parameter)).isEmpty) {
+      throw new IllegalArgumentException(s"Parameter '$parameter' is required")
+    }
+
+    parameters.get(parameter)
+  }
+
+  private def getParameter(parameter: String): Option[String] = {
```

Comment:
```suggestion
  private def getParameter(parameter: String, defaultValue: String = ""): String = {
```

così togli i `getOrElse` sotto

Reply:
alcuni default value non sono stringhe e devono essere convertiti. Vorrei evitare di ritornare cose diverse, e vorrei evitare di paassare una stringa "1000" che poi deve essere riconvertita ad intero